src/frontend/src/optimizer/plan_node/logical_agg.rs

```diff
@@ -498,6 +498,52 @@ impl LogicalAggBuilder {
         }
         None
     }
+
+    /// do some syntax check for distinct aggregates.
+    ///
+    /// TODO: we may disable this syntax check in the future because we may use another approach to
+    /// implement distinct aggregates.
+    pub fn syntax_check(&self) -> Result<()> {
+        let mut has_distinct = false;
+        let mut has_approx_count_distinct = false;
+        let mut has_order_by = false;
+        let mut single_value_has_disintct = false;
```

Comment:
This does 3 checks:
* Siblings should not have `distinct` and `approx_count_distinct` at the same time.
* Siblings should not have `distinct` and `order by` at the same time.
* `single_value(distinct)` is disallowed.

It is better to list these restrictions clearly in doc comment rather than saying "some check", alongside with reasons:

* `approx_count_distinct` is disallowed because this defeats its purpose of trading accuracy for speed. We should guide the user to improve the query with proper error message: avoid other `distinct` or just use `count(distinct)`.
* `order by` is disallowed because we are unable to produce a working plan using `expand-agg-agg` right now. This is the part that may be removed.
* `single_value(distinct)` is better disabled in binder. So does `count(distinct *)`. They are disallowed because their semantic is unclear as an agg call as of now, and we'd better not to construct them at the first place.
  * Similarly, I am also leaning towards putting the rewrite from `sum(v1 order by v2)` to `sum(v1)` in binder rather than in `rewrite_agg_call`. But `avg` rewrite is also here, so it is also okay not to move.